Sources/FoundationEssentials/_ThreadLocal.swift

```diff
@@ -46,6 +48,8 @@ struct _ThreadLocal {
             self.key = key
 #elseif canImport(WinSDK)
             key = FlsAlloc(nil)
+#elseif os(WASI)
+            key = UnsafeMutablePointer<UnsafeMutableRawPointer?>.allocate(capacity: 1)
```

Comment:
Right, it won't be freed, as well as other platforms's TLS keys won't (a key created by `pthread_key_create` here is not deallocated by `pthread_key_delete`)

And we need a double-pointer here because the `static subscript(_ key: PlatformKey) -> UnsafeMutableRawPointer?` setter requires swapping the storage pointer itself.

Reply:
Ah good point - this at least is consistent and the only use of this key is a static global anyways which shouldn't ever be deallocated. In the future we should probably update this to use `~Copyable` semantics (or something similar) to ensure we only have one copy of each key and release it in its `deinit`, but that's not WASI specific so we can do that later